osmomath/decimal_test.go

```diff
@@ -30,18 +30,106 @@ func TestDecimalTestSuite(t *testing.T) {
 	suite.Run(t, new(decimalTestSuite))
 }
 
+// BigIntDecorator is implemented by osmomath's numerical values, as Dec, BigDec, Int, Uint
+type BigIntDecorator interface {
+	BigInt() *big.Int
+}
+
+// Precision returns a precision of decimal's underlying implementation
+// Panics if decimal's underlying type is not one of BigDec, Dec, Int or Uint
+func Precision(decimal interface{}) int64 {
+	switch any(decimal).(type) {
+	case osmomath.BigDec:
+		return osmomath.PrecisionBigDec
+	case osmomath.Dec:
+		return osmomath.PrecisionDec
+	case osmomath.Int, osmomath.Uint:
+		return 0
+	default:
+		panic(fmt.Sprintf("%T type is not a decimal", decimal))
+	}
+}
+
+// mutResultToAssert is a struct with numerical values scaled to the same precision (0)
+type mutResultToAssert struct {
+	expectedResult        big.Int
+	startValue            big.Int
+	mutativeResult        big.Int
+	nonMutativeResult     big.Int
+	mutativeStartValue    big.Int
+	nonMutativeStartValue big.Int
+}
```

Comment:
note: these are non-pointers to make sure we are not comparing pointers in `assertMutResult` (even though afaik `.Equal()` function is safe to use with pointers, because it unwraps them before any assertions)